### ISSUE TITLE
Remove bits_word as an unsigned char

### ISSUE THREAD

**Issue by DavidDeSimone** [Wilfred/remacs#196]
In src/lisp.h we have 
```
/* An unsigned integer type representing a fixed-length bit sequence,
   suitable for bool vector words, GC mark bits, etc.  Normally it is size_t
   for speed, but on weird platforms it is unsigned char and not all
   its bits are used.  */
#if BOOL_VECTOR_BITS_PER_CHAR == CHAR_BIT
typedef size_t bits_word;
# define BITS_WORD_MAX SIZE_MAX
enum { BITS_PER_BITS_WORD = SIZE_WIDTH };
#else
typedef unsigned char bits_word;
# define BITS_WORD_MAX ((1u << BOOL_VECTOR_BITS_PER_CHAR) - 1)
enum { BITS_PER_BITS_WORD = BOOL_VECTOR_BITS_PER_CHAR };
#endif
verify (BITS_WORD_MAX >> (BITS_PER_BITS_WORD - 1) == 1);
```

My question for anyone in the know is: Do we care about any of these "weird platforms" in remacs, and if not, can we replace all occurrences of bits_word with just size_t? Removing this will allow us to clean up a lot of code in data.c

**Comment by jeandudey** [comment]
I currently don't make builds for these platforms nor i don't think we are able to build to them. In my humble opinion i don't care about those old unixes like Ultrix, Irix or whatever, maybe in some future we could support embedded platform and weird platforms.

I would be happy if code like this gets removed to make code clean and thereby more easy to port.

**Comment by Wilfred** [comment]
Agreed, I think it's reasonable to assume that there are 8 bits in a byte and that we're only running on 32 or 64 bit systems.

**Comment by Titan-C** [comment]
This was resolved in #201

**Comment by Wilfred** [comment]
Aha, indeed it was!
